src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/MapCode/MapCodeEndpoint.cs

```diff
@@ -82,7 +83,7 @@ public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, V
                 continue;
             }
 
-            if (!_documentContextFactory.TryCreate(mapping.TextDocument.Uri, out var documentContext))
+            if (!_documentContextFactory.TryCreate(mapping.TextDocument.DocumentUri.GetRequiredParsedUri(), out var documentContext))
```

Comment:
I wasn't expecting the `DocumentUri` change to be so viral, but this appears to be one of the reasons why - directly passing in just the URI type to helpers such as these.  

Would it be reasonable to have helpers like this take in for example a `TextDocumentIdentifier` instead?  I think that is one reason the Roslyn side changes were a bit easier

(to be clear, not suggesting it happens in this PR, but as a general improvement).